### ISSUE TITLE
specifying order in page_dir (maybe also sources)

### ISSUE THREAD

**Issue by bolding** [Fortran-FOSS-Programmers/ford#307]
I'm using page_dir to include a Users guide - like
![image](https://user-images.githubusercontent.com/11144809/79955379-0b172280-847f-11ea-83b4-e8cfe2daed77.png)

To facilitate this I need to name the .md files comprisng the users guide such that the name reflects the order I want. 

A similar issue appears with the listing of the source files - where a specific order would/could reflect the logic of the code. Keeping the directory name in the list would be the fast option.
![image](https://user-images.githubusercontent.com/11144809/79956496-83321800-8480-11ea-8e58-a03b9a76c429.png)

I don't have any ideas how to implement it though  :-)

**Comment by ecasglez** [comment]
This patch implements user-ordering for the documentation pages.

```diff
diff --git a/ford/pagetree.py b/ford/pagetree.py
index e8bcf0e..df50b47 100644
--- a/ford/pagetree.py
+++ b/ford/pagetree.py
@@ -54,6 +54,10 @@ class PageNode(object):
             self.date   = '\n'.join(md.Meta['date'])
         else:
             self.date = None
+        if 'ordered_subpage' in md.Meta:
+            self.ordered_subpages = md.Meta['ordered_subpage']
+        else:
+            self.ordered_subpages = None
         
         self.parent    = parent
         self.contents  = text
@@ -88,6 +92,12 @@ class PageNode(object):
 
     
 def get_page_tree(topdir,md,parent=None):
+
+    # In python 3.6 or newer, the normal dict is guaranteed to be ordered.
+    # However, to keep compatibility with older versions I use OrderedDict.
+    # I will use this later to remove duplicates from a list in a short way.
+    from collections import OrderedDict
+
     # look for files within topdir
     filelist = sorted(os.listdir(topdir))
     if 'index.md' in filelist:
@@ -101,9 +111,17 @@ def get_page_tree(topdir,md,parent=None):
     else:
         print('Warning: No index.md file in directory {}'.format(topdir))
         return None
-    for name in filelist:
+    if node.ordered_subpages:
+        #Merge user given files and all files in folder, removing duplicates.
+        mergedfilelist = list(OrderedDict.fromkeys(node.ordered_subpages + filelist))
+    else:
+        mergedfilelist = filelist
+
+    for name in mergedfilelist:
         if name[0] != '.' and name[-1] != '~':
-            if os.path.isdir(os.path.join(topdir,name)):
+            if not os.path.exists(os.path.join(topdir,name)):
+                raise Exception('Requested page file {} does not exist.'.format(name))
+            elif os.path.isdir(os.path.join(topdir,name)):
                 # recurse into subdirectories
                 subnode = get_page_tree(os.path.join(topdir,name),md,node)
                 if subnode: node.subpages.append(subnode)
```

With the patched file, if you want ordered subpages you can add them in file index.md using keyword ordered_subpage. Example:

```
title: title of the page
ordered_subpage: subpage3
ordered_subpage: subpage1
```

The subpages can be another md file or a folder. If it is a folder it has to contain again an index.md file where you can order the sub_sub_pages adding them using ordered_subpages in a similar way.

If some file is not added as ordered_subpage it will be added to the documentation at the end of the corresponding level. If no file is in the list, the standard ordering is used.

I will do some more tests and I will submit a pull request.

**Comment by ZedThree** [comment]
Fixed in #318
